src/emulator/storage/apis/firebase.ts

```diff
@@ -591,16 +591,21 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
   firebaseStorageAPI.delete("/b/:bucketId/o/:objectId", async (req, res) => {
     const decodedObjectId = decodeURIComponent(req.params.objectId);
     const operationPath = ["b", req.params.bucketId, "o", decodedObjectId].join("/");
+    const md = storageLayer.getMetadata(req.params.bucketId, decodedObjectId);
+
+    const rulesFiles: { before?: RulesResourceMetadata } = {};
+
+    if (md) {
+      rulesFiles.before = md.asRulesResource();
```

Comment:
Nice! Can we add a test?

Reply:
There is an [existing](https://github.com/firebase/firebase-tools/blob/5564ad1836ca91010e4b99ff78ef70f8f44bdc89/scripts/storage-emulator-integration/tests.ts#L1190) integration test for deletes. I thought I'd just add a condition on `resource` in the security rules for the test.

Reply:
In that case can we add a test for the negative case? (ie security rules denies the delite)

Reply:
Done.